client/bin/client.js

```diff
@@ -5,45 +5,102 @@ import { join, dirname } from "path";
-        // Ensure index.html is never cached
-        source: "index.html",
-        headers: [
-          {
-            key: "Cache-Control",
-            value: "no-cache, no-store, max-age=0",
-          },
-        ],
-      },
-      {
-        // Allow long-term caching for hashed assets
-        source: "assets/**",
-        headers: [
-          {
-            key: "Cache-Control",
-            value: "public, max-age=31536000, immutable",
-          },
-        ],
-      },
-    ],
-  };
+const handlerOptions = {
+  public: distPath,
+  rewrites: [{ source: "/**", destination: "/index.html" }],
+  headers: [
+    {
+      // Ensure index.html is never cached
+      source: "index.html",
+      headers: [
+        {
+          key: "Cache-Control",
+          value: "no-cache, no-store, max-age=0",
+        },
+      ],
+    },
+    {
+      // Allow long-term caching for hashed assets
+      source: "assets/**",
+      headers: [
+        {
+          key: "Cache-Control",
+          value: "public, max-age=31536000, immutable",
+        },
+      ],
+    },
+  ],
+};
 
+const requestHandler = (request, response) => {
   return handler(request, response, handlerOptions);
-});
+};
+
+const port = process.env.CLIENT_PORT || 6274;
+const INSPECTOR_SSL_CERT_PATH = process.env.INSPECTOR_SSL_CERT_PATH;
+const INSPECTOR_SSL_KEY_PATH = process.env.INSPECTOR_SSL_KEY_PATH;
+
+let server;
+
+if (INSPECTOR_SSL_CERT_PATH && INSPECTOR_SSL_KEY_PATH) {
+  // HTTPS server
+  try {
+    const options = {
+      cert: fs.readFileSync(INSPECTOR_SSL_CERT_PATH),
+      key: fs.readFileSync(INSPECTOR_SSL_KEY_PATH)
+    };
+    server = https.createServer(options, requestHandler);
+    server.on("listening", () => {
+      console.log(
+        `🔍 MCP Inspector is up and running at https://127.0.0.1:${port} 🚀🔒`,
+      );
+    });
+  } catch (error) {
+    console.error(`❌ Failed to load SSL certificates: ${error instanceof Error ? error.message : String(error)}`);
+    console.log(`🔄 Falling back to HTTP mode`);
+    server = http.createServer(requestHandler);
+    server.on("listening", () => {
+      console.log(
+        `🔍 MCP Inspector is up and running at http://127.0.0.1:${port} 🚀 (SSL fallback)`,
+      );
+    });
+  }
+} else {
+  // HTTP server (default)
+  if (!INSPECTOR_SSL_CERT_PATH && !INSPECTOR_SSL_KEY_PATH) {
+    console.log(`🔓 No SSL certificates configured - using HTTP`);
+    console.log(`💡 To enable HTTPS, set INSPECTOR_SSL_CERT_PATH and INSPECTOR_SSL_KEY_PATH environment variables`);
+  } else {
+    console.log(`⚠️  Incomplete SSL configuration:`);
+    if (!INSPECTOR_SSL_CERT_PATH) console.log(`   Missing INSPECTOR_SSL_CERT_PATH`);
+    if (!INSPECTOR_SSL_KEY_PATH) console.log(`   Missing INSPECTOR_SSL_KEY_PATH`);
+    console.log(`🔄 Using HTTP mode`);
+  }
+  
+  server = http.createServer(requestHandler);
+  server.on("listening", () => {
+    console.log(
+      `🔍 MCP Inspector is up and running at http://127.0.0.1:${port} 🚀`,
```

Comment:
This isn't right. Please merge main into your branch and resolve conflicts before proceeding.

Run the current version on `main` to see how this should be displayed. It should be `localhost` and it should contain the variables including `MCP_PROXY_AUTH_KEY` and `MCP_PROXY_PORT` if `SERVER_PORT` was set in env.

<img width="1055" alt="Screenshot 2025-07-04 at 2 28 19 PM" src="https://github.com/user-attachments/assets/7f9c0b03-4d2d-4a8f-889a-3290b1c17013" />